DevTools/HackStudio/Debugger/VariablesModel.cpp

```diff
@@ -85,9 +87,45 @@ String variable_value_as_string(const DebugInfo::VariableInfo& variable)
         return String::format("'%c' (%d)", static_cast<char>(value.value()), static_cast<char>(value.value()));
     }
 
+    if (variable.type == "bool") {
+        auto value = Debugger::the().session()->peek((u32*)variable_address);
+        ASSERT(value.has_value());
+        return (value.value() & 1) ? "true" : "false";
+    }
+
     return String::format("type: %s @ %08x, ", variable.type.characters(), variable_address);
 }
 
+void VariablesModel::set_variable_value(const GUI::ModelIndex& index, const StringView& string_value, GUI::Window* parent_window)
+{
+    auto variable = static_cast<const DebugInfo::VariableInfo*>(index.internal_data());
+
+    u32 new_value = 0;
+    bool success = false;
+
+    if (variable->type == "int") {
```

Comment:
I think we can factor out the logic here to a lambda or a static function, early return would simplify the `success` logic.
Maybe something like `evaluate_variable_value` that returns an `Optional<u32>`?

Reply:
Definitely a good idea, makes the code much more manageable.